### PR TITLE
Test changes to support github-api 1.318

### PR DESCRIPTION
**Pull request by bitwiseman**
# Description
Fix test failures due to changes in `github-api` v1.318.  

Tested manually against 1.318 and this works. 

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [x] Automated tests have been added to exercise the changes
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verify that the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

# Documentation changes
- [ ] Link to jenkins.io PR, or an explanation for why no doc changes are needed

# Users/aliases to notify

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -81,8 +81,8 @@
     </dependency>
     <dependency>
       <groupId>com.github.tomakehurst</groupId>
-      <artifactId>wiremock-jre8-standalone</artifactId>
-      <version>2.35.1</version>
+      <artifactId>wiremock-standalone</artifactId>
```
Comment (bitwiseman) [inline pom.xml:85]: Jenkins no longer supports jre8.  Use the mainline version.

**Review by joseblas — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitCheckerTest.java [hunk 2]
```diff
@@ -828,7 +840,10 @@ public void OnOverExpectedIdealOverTime() throws Exception {
                 github.getRateLimit();
             }
             // calls rateLimit() for first loop so we have to getRateLimit() for each loop
-            github.getMeta();
+            github.getLicense("MIT");
+
+            // sanity cached by github-api to one request per second
+            Thread.sleep(1000);
```
[the remark this answers is not in the document]
Comment (bitwiseman) [reply src/test/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitCheckerTest.java]: The sanity cache is rounded to the nearest second and as long is it is a different second we're okay.  I think `1000` ms in that case is fine, but I'm fine changing to `1001`.

**Comment by bitwiseman** [comment]
From the commit history you can see I've tested pre and post 1.318 and the changes work in both scenarios.  Only test changes were needed not functional changes.

**Comment by samrocketman** [comment]
> From the commit history you can see I've tested pre and post 1.318 and the changes work in both scenarios. Only test changes were needed not functional changes.

Does this mean 1.318 is unblocked from JUC or would it require another github-api release.

**Comment by MarkEWaite** [comment]
> @MarkEWaite Do we need to do a new github-api release or can we just revert the block? I've filed [jenkinsci/bom#2694](https://github.com/jenkinsci/bom/pull/2694) on the hope that reverting the block is sufficient.

No new release is needed.  Thanks for asking.  The revert is enough.

**Pull request merged**